src/main/io/ledstrip.c

```diff
@@ -938,6 +938,30 @@ static void applyLarsonScannerLayer(bool updateNow, timeUs_t *timer)
     }
 }
 
+static void applyRainbowLayer(bool updateNow, timeUs_t *timer)
+{
+    int delta = 24;
+    static int offset = 0;
+
+    if (updateNow) {
+        *timer += HZ_TO_US(60);
+    }
+
+    for (unsigned i = 0; i < ledCounts.count; i++) {
+
+        const ledConfig_t *ledConfig = &ledStripConfig()->ledConfigs[i];
+
+        if (ledGetOverlayBit(ledConfig, LED_OVERLAY_RAINBOW)) {
+            hsvColor_t ledColor;
+            ledColor.h = (offset + (i * delta)) % HSV_HUE_MAX;
```

Comment:
This calculation uses led position directly, independent of LED_OVERLAY_RAINBOW. It should be probably using index within rainbow group(see other effects, like larson)

Reply:
Ahh thank you @ledvinap good catch! I will look to update the code to keep track of only the pixels in the rainbow overlay!